.github/workflows/go-test.yml

```diff
@@ -0,0 +1,26 @@
+name: go-test
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
```

Comment:
Nit: would be good to run tests for PRs not against `main` branch

Reply:
Good point. I tried removing that filter entirely here, which it seems from the docs should mean to always run the tests agains PRs.

Reply:
Update: it seems to be working, because I see the tests running under "Checks" below.